### PR TITLE
Cherry picked from Azure theme 8.0 danger button commit

### PR DESCRIPTION
**Pull request by Jacqueline-ms**
Cherry picked from 8.0 #25383 into the 7.0 branch 
## Previous Behavior

image

## New Behavior

Danger button
Denoted by <PrimaryButton className="danger" /> or <DefaultButton className="danger" /> only in Azure Theme.

**Light theme (all states)**
rest
image

hover
image

pressed / focus
image

Dark theme
image

**High contrast themes follow the same rules as regular buttons, according to Azure KO SDK**
High contrast light (rest & hover)
image

High contrast dark (rest & hover)
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">7.0 Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/BaseButton-mount_base.html">904</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/BaseButton-mount.html">914</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Breadcrumb-mount_base.html">42906</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Breadcrumb-mount.html">43267</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Checkbox-mount_base.html">1572</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Checkbox-mount.html">1553</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/CheckboxBase-mount_base.html">1268</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/CheckboxBase-mount.html">1307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/ChoiceGroup-mount_base.html">4939</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/ChoiceGroup-mount.html">4949</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/ComboBox-mount_base.html">961</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/ComboBox-mount.html">890</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/CommandBar-mount_base.html">7918</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/CommandBar-mount.html">7917</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/ContextualMenu-mount_base.html">14339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/ContextualMenu-mount.html">14515</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DefaultButton-mount_base.html">1145</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DefaultButton-mount.html">1122</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DetailsRow-mount_base.html">3745</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DetailsRow-mount.html">3758</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DetailsRowFast-mount_base.html">3746</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DetailsRowFast-mount.html">3725</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DetailsRowNoStyles-mount_base.html">3567</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DetailsRowNoStyles-mount.html">3587</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Dialog-mount_base.html">2301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Dialog-mount.html">2237</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DocumentCardTitle-mount_base.html">1839</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/DocumentCardTitle-mount.html">1846</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Dropdown-mount_base.html">2497</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Dropdown-mount.html">2520</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/FocusTrapZone-mount_base.html">1749</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/FocusTrapZone-mount.html">1717</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/FocusZone-mount_base.html">1859</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/FocusZone-mount.html">1844</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/IconButton-mount_base.html">1739</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/IconButton-mount.html">1726</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Label-mount_base.html">333</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Label-mount.html">349</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Layer-mount_base.html">1952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Layer-mount.html">1971</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Link-mount_base.html">442</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Link-mount.html">442</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/MenuButton-mount_base.html">1486</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/MenuButton-mount.html">1475</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/MessageBar-mount_base.html">2070</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/MessageBar-mount.html">2097</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Nav-mount_base.html">3294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Nav-mount.html">3338</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/OverflowSet-mount_base.html">1413</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/OverflowSet-mount.html">1411</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Panel-mount_base.html">1483</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Panel-mount.html">1455</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Persona-mount_base.html">964</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Persona-mount.html">948</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Pivot-mount_base.html">1492</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Pivot-mount.html">1478</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/PrimaryButton-mount_base.html">1292</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/PrimaryButton-mount.html">1302</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Rating-mount_base.html">7839</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Rating-mount.html">7886</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SearchBox-mount_base.html">1259</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SearchBox-mount.html">1250</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Shimmer-mount_base.html">2568</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Shimmer-mount.html">2590</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Slider-mount_base.html">1505</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Slider-mount.html">1526</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SpinButton-mount_base.html">5103</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SpinButton-mount.html">5092</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Spinner-mount_base.html">425</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Spinner-mount.html">417</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SplitButton-mount_base.html">3192</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SplitButton-mount.html">3210</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Stack-mount_base.html">511</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Stack-mount.html">504</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/StackWithIntrinsicChildren-mount_base.html">1696</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/StackWithIntrinsicChildren-mount.html">1703</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/StackWithTextChildren-mount_base.html">4834</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/StackWithTextChildren-mount.html">4774</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SwatchColorPicker-mount_base.html">10656</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/SwatchColorPicker-mount.html">10615</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/TagPicker-mount_base.html">2905</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/TagPicker-mount.html">2898</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/TeachingBubble-mount_base.html">51186</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/TeachingBubble-mount.html">51604</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Text-mount_base.html">414</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Text-mount.html">415</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/TextField-mount_base.html">1407</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/TextField-mount.html">1370</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Toggle-mount_base.html">822</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/Toggle-mount.html">837</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/button-mount_base.html">111</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25892/perf-test/button-mount.html">122</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Review by robtaft-ms — APPROVED** [review]
This looks great, thank you Jackie!

**Review by robtaft-ms — APPROVED** [review]
This looks good now, thank you!

File: change/@uifabric-azure-themes-819da265-3f92-400b-99db-6533f3df58ef.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Cherry picked from Azure theme 8.0 danger button commit",
```
Comment (khmakoto) [inline change/@uifabric-azure-themes-819da265-3f92-400b-99db-6533f3df58ef.json:3]:
```suggestion
  "comment": "feat: Add styles for indeterminate checkbox and danger button.",
```

**Review by khmakoto — APPROVED** [review]

File: change/@uifabric-azure-themes-819da265-3f92-400b-99db-6533f3df58ef.json (hunk 1 above)
Comment (Jacqueline-ms) [reply change/@uifabric-azure-themes-819da265-3f92-400b-99db-6533f3df58ef.json:3]: I'll be sure to use more specific yarn change comments, thanks!

**Comment by fabricteam** [comment]
## 🕵 FluentUI-v7 No visual regressions between this PR and main
<div id="vrtCommentFluentUI-v7"/>

**Pull request merged**